src/Anonymization/Anonymizer/Core/FloatAnonymizer.php

```diff
@@ -23,27 +27,109 @@ class FloatAnonymizer extends AbstractAnonymizer
      */
     public function anonymize(Update $update): void
     {
-        if (!($this->options->has('min') && $this->options->has('max'))) {
-            throw new \InvalidArgumentException("You should provide 2 options (min and max) with this anonymizer");
+
+        $precision = $this->options->get('precision', 2);
+        if ($precision <= 0) {
+            throw new \InvalidArgumentException("'precision' should be greater than 0.");
         }
+        $precision = 10 ** $precision;
+
+        if ($this->options->has('min') && $this->options->has('max')) {
+            $min = (float) $this->options->get('min');
+            $max = (float) $this->options->get('max');
+            if ($min >= $max) {
+                throw new \InvalidArgumentException("'max' should be greater than 'min'.");
+            }
+
+            $this->anonymizeWithMinAndMax($update, $precision, $min, $max);
+        } elseif ($this->options->has('delta')) {
+            $delta = (float) $this->options->get('delta');
```

Comment:
You maybe should check if `'percent'` is present too, and raise an exception if both are specified.

Reply:
yep, what I'm gonna say is not coherent with my last comment but in fact, I thought to refin this with #97